components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -2864,6 +2864,14 @@ public static boolean jsFunction_addDocumentation(Context cx, Scriptable thisObj
 
         APIIdentifier apiId = new APIIdentifier(APIUtil.replaceEmailDomain(providerName), apiName, version);
         Documentation doc = new Documentation(getDocType(docType), docName);
+        APIProvider apiProvider = getAPIProvider(thisObj);
+
+        //add documentation is allowed only if document name does not already exist for this api
+        if (apiProvider.isDocumentationExist(apiId, docName)) {
+            handleException("Error occurred while adding the document. " + docName +
+                    " already exists for API " + apiName + "-" + version);
```

Comment:
If you run findbugs on this, you will see the following warning :)

Method passes constant String of length 1 to character overridden method
This method passes a constant literal String of length 1 as a parameter to a method, that exposes a similar method that takes a char. It is simpler and more expedient to handle one character, rather than a String.
Instead of making calls like: String myString = ... if (myString.indexOf("e") != -1) { int i = myString.lastIndexOf("e"); System.out.println(myString + ":" + i); //the Java compiler will use a StringBuilder internally here [builder.append(":")] ... return myString.replace("m","z"); } Replace the single letter Strings with their char equivalents like so: String myString = ... if (myString.indexOf('e') != -1) { int i = myString.lastIndexOf('e'); System.out.println(myString + ':' + i); //the Java compiler will use a StringBuilder internally here [builder.append(':')] ... return myString.replace('m','z'); }